<!DOCTYPE html><html><head><link href="limbr.a7bfb861.css" rel="stylesheet" type="text/css"><meta charset="utf-8"><link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet"></head><body> <div class="topnav"> <a href="index.html">LIMBR</a> <div class="topnav-right"> <a class="active" href="about.html">About</a> <a href="#Data">Data</a> <a href="#git">git</a> <a href="#contact">Contact</a> </div> </div> <div id="root" class="tab"> <img src="Figure1SCN1A.cc98b884.jpg" alt="Figure1LIMBR" width="75%" height="75%" class="center"> <figcaption> Figure 1 (from Hayeck et al.)<sup>1</sup> Relative missense variation versus total variation across domains. Regions (domains) are plotted with the number of total SNV variants in each coding region on the x-axis versus the number of missense variants on the y- axis. The genome wide average missense variation versus total variation is plotted as a black solid line (A). Highlighted in blue are the SCN1A domains as an example. The offset average gene level trend for SCN1A is plotted as a blue dotted line (B) and can be seen more clearly in the exploded panel. Fitting a Bayesian hierarchical model allows for sharing of information across sub-regions, pulling the sub-region level terms towards the genic average. </figcaption> <h2>LIMBR</h2> <p> Localized Intolerance Model using Bayesian Regression (LIMBR) is a sub-regional (domains or exons) genic intolerance score. We fit a Bayesian hierarchical model explicitly characterizing depletion in functional variation at both the gene and sub-regional level. Figure 1 from our manuscript<sup>1</sup> depicts the approximate geometric interpretation of this approach, the model was regressed on missense variation versus all variation. </p> <h3>Data</h3> <p> The data on this website corresponds to our updated fit of the model on genome Aggregation Database (<a href="https://gnomad.broadinstitute.org">gnomAD</a>) version v2.1 using 125,748 whole exome sequences. Two sets of scores are calculated by fitting different definitions of sub-regions across genes, once with genic sub-regions defined by exon boundaries and then again with sub-regions defined by functional domains, both using the Conserved Domain Database (<a href="https://www.ncbi.nlm.nih.gov/cdd/">CDD</a>).<sup>2,3</sup> The filtered data from gnomAD first had to go through ‘PASS’ criteria, or in this version we also allowed for SEGDUP variants to be included, then further restricted to regions with at least 10x coverage in at least 70% of the samples. Additionally, any genes without any variation were excluded from the analysis. </p> <h3>Model</h3> <p> The number of missense variants within the ith sub-region of gene j, y<sub>ij</sub>, as a function of the total number of variants within the ith sub-region of gene j, x<sub>ij</sub>, through the following regression model, </p> <p class="equation"> y<sub>ij</sub> = μ+α<sub>j</sub> + α<sub>i(j)</sub> + β<sub>1</sub>x<sub>ij</sub> + e<sub>0ij</sub>, </p> <p> We model α<sub>j</sub> and α<sub>i(j)</sub> as random effects within a Bayesian framework. We assume standard normal priors for α<sub>j</sub> and α<sub>i(j)</sub> but allow for separate prior variances for α<sub>i(j)</sub> for each gene j. We begin by assuming an inverse gamma prior for the variances. Specifically, we choose α<sub>j</sub>~N(0,σ<sup>2</sup> ) with hyper-parameters σ<sup>2</sup>~InvGamma( ϵ,ϵ) and ϵ~Uniform(δ,c), where δ is a small positive constant and c is a large positive constant to induce a diffuse prior. For the sub-region parameters, we use a similar structure but with a separate variance for each gene, i.e., we choose α<sub>i(j)</sub>~N(0,σ<sub>j</sub><sup>2</sup> ) with hyper-parameters σ<sub>j</sub><sup>2</sup>~InvGamma( ϵ<sub>j</sub>,ϵ<sub>j</sub>) and ϵ<sub>j</sub>~Uniform(δ,c). Note that by allowing for a gene-level variance, the α<sub>ij</sub>s can be shrunken back to the gene level intolerance when there are no large differences between sub-region or when data is sparse. This will decrease the variability of the α<sub>i(j)</sub>s, leading to more stable intolerance estimation. </p> <p> To improve the ergodicity, α<sub>i(j)</sub> was set to zero for genes with 2 or fewer sub-regions, this is effectively just collapsing genes with only 2 sub-regions eliminating any inflated within versus across chain variance. Further, it is known that the hierarchical model can be augmented, sometimes referred to as noncentral<sup>4–6</sup> or ancillary augmentation<sup>7</sup>, and similar methods are known to improve performance.<sup>8,9</sup> So, we introduced an additional hyper parameter v~N(0,1) for α<sup>*</sup><sub>ij</sub>=vσ<sub>j</sub> to reduce autocorrelation. </p> <p class="equation"> y<sub>ij</sub> = μ+α<sub>j</sub> + α<sup>*</sup><sub>i(j)</sub> + β<sub>1</sub>x<sub>ij</sub> + e<sub>0ij</sub>, </p> <p> By introducing an auxiliary hyper parameter, the conditional variance structure is maintained while decoupling the random variables we wish to make inference on, Var(α<sub>ij</sub>│σ<sub>j</sub><sup>2</sup> )=Var( α<sup>*</sup><sub>i(j)</sub>│v,σ<sub>j</sub><sup>2</sup> ). The final score that is used for the analysis is the posterior mode of the combined genic and sub-region terms. The hierarchical model allows information to be shared across sub-regions, stabilizing intolerance estimates. A burn in of 1,000 with an additional 10,000 steps across 5 chains was run for both domains and exons. </p> <p> For more details please see our article: <a href="https://www.sciencedirect.com/science/article/pii/S0002929718305032">Improved pathogenic variant localization via a hierarchical model of sub-regional intolerance</a> </p> <h3>Reference</h3> <p> If you find the information or resources from this website useful in your research please cite our paper. </p> <p> Hayeck, T. J., Stong, N., Wolock, C. J., Copeland, B., Kamalakaran, S., Goldstein, D. B., & Allen, A. S. (2019). Improved pathogenic variant localization via a hierarchical model of sub-regional intolerance. The American Journal of Human Genetics, 104(2), 299-309. </p> <h3>Bibliography</h3>  <ol> <li> Hayeck, T. J. et al. Improved Pathogenic Variant Localization via a Hierarchical Model of Sub-regional Intolerance. Am. J. Hum. Genet. 1–11 (2019). doi:10.1016/j.ajhg.2018.12.020 </li> <li> Marchler-Bauer, A. et al. CDD: Conserved domains and protein three-dimensional structure. Nucleic Acids Res. 41, 348–352 (2013). </li> <li>... add the rest in XXX</li> <li> Gussow, A. B., Petrovski, S., Wang, Q., Allen, A. S. & Goldstein, D. B. The intolerance to functional genetic variation of protein domains predicts the localization of pathogenic mutations within genes. Genome Biol. 17, 9 (2016). </li> <li> Papaspiliopoulos, O. & Roberts, G. O. Non-Centered Parameterisations for Hierarchical Models and Data Augmentation. Bayesian Stat. 7, 307–326 (2003). </li> <li> Papaspiliopoulos, O. & Roberts, G. Stability of the Gibbs sampler for Bayesian hierarchical models. Ann. Stat. 36, 95–117 (2008). </li> <li> Betancourt, M. A General Metric for Riemannian Manifold Hamiltonian Monte Carlo. 327–334 (2013). </li> <li> Yu, Y. & Meng, X.-L. To Center or Not to Center: That Is Not the Question—An Ancillarity–Sufficiency Interweaving Strategy (ASIS) for Boosting MCMC Efficiency. J. Comput. Graph. Stat. 20, 531–570 (2011). </li> <li> Duan, L. L., Johndrow, J. E. & Dunson, D. B. Scaling up Data Augmentation MCMC via Calibration. (2017). </li> <li> Liu, J. S. The collapsed Gibbs sampler with applications to a gene regulation problem. J. Amer. Stat. Assoc 89, 958–966 (1994). </li> </ol>  </div> </body></html>